### PR TITLE
fix/PSD-3993_http_session_timeout

### PR DESCRIPTION
**Pull request by tahir-khalid**
## Description

Long HTTP Session Timeout - A review of the device identified that an excessive HTTP timeout was configured.

Set devise session timeout to 30 minutes.

### CODE REVIEW

File: spec/features/sign_in_spec.rb [hunk 1]
```diff
@@ -192,7 +192,7 @@ def otp_code
     visit investigation_path(investigation)
     expect(page).not_to have_text("You need to sign in or sign up before continuing.")
 
-    travel_to 24.hours.from_now do
+    travel_to 31.hours.from_now do
```
Comment (meeravnagesh) [inline spec/features/sign_in_spec.rb:195]: check for 29 minutes and 31 minutes

**Review by meeravnagesh — CHANGES_REQUESTED** [review]

File: spec/features/sign_in_spec.rb (hunk 1 above)
Comment (meeravnagesh) [reply spec/features/sign_in_spec.rb:195]: the code is in hours

File: spec/features/redirect_after_2fa_spec.rb [hunk 2]
```diff
@@ -43,27 +40,26 @@
 
   context "when making a different request" do
     it "redirects user back to the last page they were on" do
-      sign_in(admin)
       visit "/teams/#{team.id}/invitations/new"
```
Comment (meeravnagesh) [inline spec/features/redirect_after_2fa_spec.rb:43]: you missed removing this line

**Review by meeravnagesh — CHANGES_REQUESTED** [review]

File: spec/features/sign_in_spec.rb [hunk 3]
```diff
@@ -192,9 +192,17 @@ def otp_code
...
+      sign_in_with_otp_code
       visit investigation_path(investigation)
+
       expect(page).not_to have_text("Your session expired. Please sign in again to continue.")
+      expect(page).to have_content("From here you can find all the information about this notification. If you have editing rights for this notification, you can add products, businesses and supporting information.")
+    end
+
+    travel_to 60.minutes.from_now do
+      visit investigation_path(investigation)
+      expect(page).to have_text("If you don’t have an account, ask someone in your team to send you an invite")
```
[the remark this answers is not in the document]
Comment (tahir-khalid) [reply spec/features/sign_in_spec.rb:205]: No not during my testing which is why I was having issues getting the tests completed unexpected results so I adjusted the tests.

File: spec/features/invite_user_spec.rb [hunk 4]
```diff
@@ -95,14 +95,22 @@
...
-        sign_in(user)
-      end
-    end
+    scenario "sends the invite after session expiry and secondary authentication" do
+      sign_in(user)
+      expect(page).to have_current_path("/")
+
+      user.update!(current_sign_in_at: 31.minutes.ago)
+
+      Capybara.current_session.reset!
```
Comment (meeravnagesh) [inline spec/features/invite_user_spec.rb:104]: what is the reason for session reset ?

File: spec/features/redirect_after_2fa_spec.rb (hunk 2 above)
Comment (tahir-khalid) [reply spec/features/redirect_after_2fa_spec.rb:43]: TODO: remove the visit line and retest

**Comment by alan-at-work** [comment]
once updated with new timeouts, update the tests too and make sure we have all as **green**

File: spec/features/invite_user_spec.rb [hunk 5]
```diff
@@ -96,7 +96,7 @@
     let(:user) { create(:user, :activated, :team_admin, team:, has_viewed_introduction: true) }
 
     before do
-      travel_to(4.hours.ago) do
```
Comment (meeravnagesh) [inline spec/features/invite_user_spec.rb:99]: do we need this change now , 4 hours would also work right ?

**Review by meeravnagesh — CHANGES_REQUESTED** [review]

File: spec/features/invite_user_spec.rb (hunk 5 above)
Comment (tahir-khalid) [reply spec/features/invite_user_spec.rb:99]: Reverted

**Review by meeravnagesh — APPROVED** [review]

**Pull request merged**